### ISSUE TITLE
auto-save-visited-mode clears echo area after 5 seconds

### ISSUE THREAD

**Issue by meedstrom** [hlissner/doom-emacs#5525]
### What did you expect to happen?

Enabling auto-save-visited-mode should not change how the echo area behaves.

### What actually happened?

With it enabled, echo area is cleared about 5 seconds after any message. That's obviously troublesome when there is important info.

### Describe your attempts to resolve the issue

Since the problem does not exist in `emacs -Q` (my initial bug report is at http://debbugs.gnu.org/cgi/bugreport.cgi?bug=50224), I bisected my init files and tracked it down to the `:ui doom` module.

### Steps to reproduce

1. Enable the :ui doom module
2. Start Emacs
3. `M-x auto-save-visited-mode RET`
4. You should see the message "Auto-save-visited-mode enabled"
5. Wait about 5 seconds
6. It's gone

### System Information

https://pastebin.com/DfhCuPq3

**Comment by hlissner** [comment]
I can reliably reproduce this, but I do not understand why it happens and why it seems to be connected to the `doom-themes` package. I'll keep poking around.

**Comment by meedstrom** [comment]
I hope this unearths a bug with save-some-buffers. Some ideas

- doom-themes does something to print a message?
- doom-themes overrides a defvar/defun?
- doom-themes modifies how save-some-buffers work thru changing a variable like save-some-buffers-default-predicate?
- save-some-buffers tries to save a (temp?) buffer?
- save-some-buffers does something crazy with custom-theme-load-path?

As far as I can see, save-some-buffers first uses dolist on a subset of buffers to save unconditionally, then uses map-y-or-n-p over `(buffer-list)`, which may result in a series of prompts, and then tries to save abbrevs. 

Unfortunately ripgrepping in doom-themes for words like "save" or "abbrev", "buffer" or "timer" didn't tell me much. 

Maybe we could write shortened alternatives to `save-some-buffers` and see if the problem goes away. Bingo! The issue goes away if you delete the `map-y-or-n-p` chunk in `save-some-buffers`.  

That results in an error, but it goes away if you delete the subsequent sexps too (which are looking for values for the lexical `files-done` and `queried`), so I think it's incidental. 

Changing the lambdas passed to `map-y-or-n-p` to return nil doesn't influence the result, and changing it so it doesn't map over `(buffer-list)` but simply two buffers `(list (car (buffer-list)) (cadr (buffer-list)))` doesn't either, so it must be in the source to that function in the first place, unfortunately C code.

**Comment by meedstrom** [comment]
Yup... you can reproduce the issue without auto-save-visited-mode. Simply eval this.

```
(run-with-timer 1 nil
 (lambda () 
 (map-y-or-n-p (lambda (a) nil) (lambda (b) nil) nil)))
```

**Comment by hlissner** [comment]
> `map-y-or-n-p`

Ah, then it wasn't `doom-themes`, but `solaire-mode` instead, due [to this hack](https://github.com/hlissner/emacs-solaire-mode/blob/master/solaire-mode.el#L516-L531). Removing it fixes the issue.

Emacs just can't give me a break. I'll experiment and see if that advice to behave.

**Comment by hlissner** [comment]
Throw this in `~/.doom.d/config.el` and see if it solves the issue for you:

```elisp
(defadvice! fix-solaire-mode-fix-minibuffer (&optional unset)
  :override #'solaire-mode-fix-minibuffer
  (dolist (buf '(" *Minibuf-0*" " *Minibuf-1*"
                 " *Echo Area 0*" " *Echo Area 1*"))
    (with-current-buffer (get-buffer-create buf)
      (if (or unset (not solaire-global-mode))
          (solaire-mode -1)
        (setq-local evil-buffer-regexps '((".")))
        (when (= (buffer-size) 0)
          (insert " "))
        (add-hook 'kill-buffer-query-functions #'ignore nil 'local)
        (solaire-mode +1)))))
```

**Comment by meedstrom** [comment]
Yes, seems to be solved.
